app/Jobs/Bank/ProcessBankTransactionsNordigen.php

```diff
@@ -0,0 +1,183 @@
+        $this->from_date = $bank_integration->from_date ?: now()->subDays(90);
+        $this->company = $this->bank_integration->company;
+    }
+
+    /**
+     * Execute the job.
+     *
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        if ($this->bank_integration->integration_type != BankIntegration::INTEGRATION_TYPE_NORDIGEN)
+            throw new \Exception("Invalid BankIntegration Type");
+
+        if (!(config('ninja.nordigen.secret_id') && config('ninja.nordigen.secret_key')))
+            throw new \Exception("Missing credentials for bank_integration service nordigen");
+
+        $this->nordigen = new Nordigen();
+
+        set_time_limit(0);
+
+        nlog("Nordigen: Processing transactions for account: {$this->bank_integration->account->key}");
+
+        // UPDATE ACCOUNT
+        try {
+            $this->updateAccount();
+        } catch (\Exception $e) {
+            nlog("Nordigen: {$this->bank_integration->nordigen_account_id} - exited abnormally => " . $e->getMessage());
+
+            $content = [
+                "Processing transactions for account: {$this->bank_integration->nordigen_account_id} failed",
+                "Exception Details => ",
+                $e->getMessage(),
+            ];
+
+            $this->bank_integration->company->notification(new GenericNinjaAdminNotification($content))->ninja();
+
+            throw $e;
+        }
+        if (!$this->nordigen_account)
+            return;
+
+        // UPDATE TRANSACTIONS
+        try {
+            $this->processTransactions();
+        } catch (\Exception $e) {
+            // reset from_date in case this was the error (self-heal) and perform a max sync of data we can fetch (next-time) @todo we should analyze the error for this
+            $this->bank_integration->from_date = now()->subDays(90);
+            $this->bank_integration->save();
+
+            nlog("Nordigen: {$this->bank_integration->nordigen_account_id} - exited abnormally => " . $e->getMessage());
+
+            $content = [
+                "Processing transactions for account: {$this->bank_integration->nordigen_account_id} failed",
+                "Exception Details => ",
+                $e->getMessage(),
+            ];
+
+            $this->bank_integration->company->notification(new GenericNinjaAdminNotification($content))->ninja();
+
+            throw $e;
+        }
+
+        // Perform Matching
+        BankMatchingService::dispatch($this->company->id, $this->company->db);
+    }
+
+    private function updateAccount()
+    {
+        if (!$this->nordigen->isAccountActive($this->bank_integration->nordigen_account_id)) {
+            $this->bank_integration->disabled_upstream = true;
+            $this->bank_integration->save();
+            $this->stop_loop = false;
+            nlog("Nordigen: account inactive: " . $this->bank_integration->nordigen_account_id);
+            // @turbo124 @todo send email for expired account
```

Comment:
I'll add this to my todo, do you know if there is a standard block of text that nordigen has for this? Or do we need to create this content ourselves?

Reply:
I did not found any resources, that nordigen handles such cases, we have to send it ourselfes I guess. Maybe we should also add a field in the bank_integration, giving the frontend information, how long the integration will probably last, so the frontend can display a hint to the user to refresh connection a few days before.

Requisitions contain access for 90 days by default.